.github/workflows/integration.yml

```diff
@@ -14,7 +14,14 @@
 
 # Run secret-dependent integration tests only after approval
 name: Integration tests
-on: pull_request
+
+on:
+  push:
+    branches:
+      - main # The e2e coverage is required to be run on main branch to get the coverage report
+  pull_request:
+    branches:
+      - main
```

Comment:
I had thought we disabled this at one point due to token usage from running integration tests before/after PR merge?

I wonder if there's anyway to use artifacts so we can upload it from `main.yml` without re-running it?

Reply:
I examined the file for blame and found that it had never been pushed to the main section. Can we enable it and see if it is causing any problems?

Maintaining the state from a pull request to a main branch merge can be complicated. I'm not sure how to do it.